src/uu/seq/src/seq.rs

```diff
@@ -332,6 +336,44 @@ fn fast_print_seq(
     Ok(())
 }
 
+#[cfg(unix)]
+static SIGPIPE_WAS_IGNORED: AtomicBool = AtomicBool::new(false);
+
+#[cfg(unix)]
+/// # Safety
+/// This function runs once at process initialization and only observes the
+/// current `SIGPIPE` handler, so there are no extra safety requirements for callers.
+unsafe extern "C" fn capture_sigpipe_state() {
+    use nix::libc;
+    use std::{mem::MaybeUninit, ptr};
+
+    let mut current = MaybeUninit::<libc::sigaction>::uninit();
+    if unsafe { libc::sigaction(libc::SIGPIPE, ptr::null(), current.as_mut_ptr()) } == 0 {
+        let ignored = unsafe { current.assume_init() }.sa_sigaction == libc::SIG_IGN;
```

Comment:
are you sure it is correct ?
it- should compare sa_handler field instead, as sa_sigaction is used for SA_SIGINFO handlers no ?

Reply:
I guess the way that I'm validating this is by using this approach and then seeing if it can be applied to fix all of the SIGPIPE related tests: https://github.com/uutils/coreutils/pull/9646 

I'll do a bit more research but I guess I don't currently have a test case where this doesn't match the expected behaviour